### DISCUSSION TITLE
subscribe object propertiesas rxjs operators

### DISCUSSION THREAD

**Discussion by Judimax** [ReactiveX/rxjs#6521] [Ideas / Feature request]
Greetings to all 
everyone familar with the Observer properties (next,complete,error)
wouldn't it be nice to have them as rxjs operators, it would make our code look much cleaner

**Comment by voliva** [comment]
Do you have any example? It's a bit hard to picture without one

**Reply by Judimax** [reply]
![image](https://user-images.githubusercontent.com/21322737/125426119-f7802e82-8000-4928-b4ce-7a38ef558f60.png)

**Reply by voliva** [reply]
Personally I rather have it in subscribe - As it's what executes the observable and looks for updates.

I like to see the operators that you'd put inside `.pipe` as functions that somehow transform the observable or use the observable without directly subscribing on it. If you need to make a side effect on one of the events we have tap:

```ts
.pipe(
  tap({
    next: () => {}
  })
)
```

But having to add in extra operators to just subscribe feels unnecessary and a bit harder to read to me.

**Comment by Judimax** [comment]
```ts
// @ts-nocheck
import {of} from 'rxjs';
import {next,error,complete,subscribe} from 'rxjs/operators';


of({})
.pipe(
    timeout(2500),
    concatMap(() => of(1)),
    exhaust(),
    next(()=>{}),
    error(()=>{}),
    complete(()=>{}),
)
.subscribe()

ajax("www.google.com")
.pipe(
    concatMap(() => of(1)),
    exhaust(),    
    next(()=>{}),
    error(()=>{}),
    complete(()=>{}),
)
.subscribe()


defer(()=>{return from(myPromise)})
.pipe(
    concatMap(() => of(1)),
    mergeMap(() => of(1)),
    error(()=>{}),
)
.subscribe()


http.post(
    endpoint,
    {large JSON body},
    {many headers and params}
)
.pipe(
    takeUntil(()=>{return true}),
    complete(()=>{}),
)
.subscribe()

```
would be neater than how we would regularly treat the Observer object properties
